src/tests/cmocka/common_utils.c

```diff
@@ -0,0 +1,66 @@
+/*
+    Authors:
+        Sumit Bose <sbose@redhat.com>
+
+    Copyright (C) 2022 Red Hat
+
+    SSSD tests: common helpers
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "tests/cmocka/common_mock.h"
+
+static struct CMUnitTest select_test(const struct CMUnitTest tests[],
+                                     size_t test_count, const char *name)
+{
+    size_t c;
+    struct CMUnitTest empty = { 0 };
```

Comment:
As you are comparing the output of this function with `NULL` please also initialize it to `NULL`

Reply:
Hi,

please note that not the returned value is checked, but a member of the returned struct. `{ 0 }` is a common way to initialized a struct with so that every member is `0` or `NULL` or `false` depending on the type, a similar way would be to use `memset()` but this would be only done at runtime while `{ 0 }` is already done at compile time. I would prefer to keep this generic way instead of initializing each member individually because the generic version must not be update if new members will be added to the struct.

bye,
Sumit

Reply:
Good explanation.